src/bootloader/bootloaderlite.cc

```diff
@@ -0,0 +1,85 @@
+#include "bootloaderlite.h"
+#include "bootloader/bootloader.h"
+
+#include <fcntl.h>
+#include <sys/reboot.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include <boost/filesystem.hpp>
+
+#include "rollbacks/factory.h"
+#include "storage/invstorage.h"
+#include "utilities/exceptions.h"
+
+BootloaderLite::BootloaderLite(BootloaderConfig config, INvStorage& storage)
+    : Bootloader(config, storage),
+      config_(std::move(config)),
+      storage_(storage),
+      rollback_(RollbackFactory::makeRollback(config_.rollback_mode)) {
+  reboot_sentinel_ = config_.reboot_sentinel_dir / config_.reboot_sentinel_name;
+  reboot_command_ = config_.reboot_command;
+
+  if (!Utils::createSecureDirectory(config_.reboot_sentinel_dir)) {
+    LOG_WARNING << "Could not create " << config_.reboot_sentinel_dir << " securely, reboot detection support disabled";
+    reboot_detect_supported_ = false;
+    return;
+  }
+  reboot_detect_supported_ = true;
+}
+
+void BootloaderLite::setBootOK() const { rollback_->setBootOK(); }
+
+void BootloaderLite::updateNotify() const { rollback_->updateNotify(); }
+
+bool BootloaderLite::supportRebootDetection() const { return reboot_detect_supported_; }
+
+bool BootloaderLite::rebootDetected() const {
```

Comment:
It looks like most of the methods are copy-paste from libaktualizr's Bootloader. If so, then they can be removed from here since BootloaderLite inherits Bootloader.

Reply:
what this means is that the intention - as it was via the abstract interface- is to not reuse anything from the base class.

Now, apparently this is now not the objective (I am sure I misunderstood); so that being the case, I'd much rather go back to having an abstract class which is the way it should be....so clients can deliver their own implementations.